File: examples/python/detect_empty_backup_file_cron.py
#!/usr/bin/env python3
"""
Detect Empty Backup File Cron
Problem: Backup job completes but creates empty file
Solution: Send file size in payload, validate in DeadManPing panel
Documentation: https://deadmanping.com/detect-empty-backup-file-cron
"""

import os
import subprocess
import requests
from datetime import datetime

MONITOR_ID = "YOUR_MONITOR_ID"  # Replace with your actual monitor ID
backup_file = f"/backups/db-{datetime.now().strftime('%Y%m%d')}.sql"

# Run backup
subprocess.run(["pg_dump", "mydb"], stdout=open(backup_file, "w"))

# Get file size
file_size = os.path.getsize(backup_file) if os.path.exists(backup_file) else 0

# Always ping with file size in payload
# In DeadManPing panel: set validation rule "size" > 0 (or >= 1024 for minimum size)
# Panel will automatically detect if size is 0 or too small
requests.post(f"https://deadmanping.com/api/ping/{MONITOR_ID}?size={file_size}")

File: examples/nodejs/detect_empty_backup_file_cron.js
#!/usr/bin/env node
/**
 * Detect Empty Backup File Cron
 * Problem: Backup job completes but creates empty file
 * Solution: Send file size in payload, validate in DeadManPing panel
 * Documentation: https://deadmanping.com/detect-empty-backup-file-cron
 */

const fs = require('fs');
const { execSync } = require('child_process');
const https = require('https');

const MONITOR_ID = 'YOUR_MONITOR_ID';  // Replace with your actual monitor ID
const backupFile = `/backups/db-${new Date().toISOString().split('T')[0].replace(/-/g, '')}.sql`;

// Run backup
execSync(`pg_dump mydb > ${backupFile}`);

// Get file size
const stats = fs.statSync(backupFile);

// Always ping with file size in payload
// In DeadManPing panel: set validation rule "size" > 0 (or >= 1024 for minimum size)
// Panel will automatically detect if size is 0 or too small
https.request(`https://deadmanping.com/api/ping/${MONITOR_ID}?size=${stats.size}`, { method: 'POST' }).end();
